proxy/ParentSelection.cc

```diff
@@ -126,6 +127,9 @@ ParentConfigParams::findParent(HttpRequestData *rdata, ParentResult *result, uns
   tablePtr->Match(rdata, result);
   rec = result->rec;
 
+  // Set the result url string to store the URL request that initiated this
+  result->url = rdata->get_string();
```

Comment:
How do you know this pointer is stable?

Reply:
By code inspection, the pointer returned by rdata->get_string() is stable because it is allocated in hdr->url_string_get() which is called from rdata->get_string().

So as the code in this PR stands, things are stable, but you need to add logic to free the result->url.

Reply:
getting back to this, Im not sure if there is a good one specific free location, so I figure instead I have to take the approach of storing the rdata ptr and just do get_string/free when I need to use it. So Im testing with that now